### PR TITLE
Change all remaining utilities tests and three main tests to TypeScript

### PR DESCRIPTION
**Pull request by Nelias**
A GIF or MEME to give some spice of the internet

## Description

Seven files in total have been changed to TypeScript, some of them required changes in the old test logic.

## Motivation and Context

[Migrate to TypeScript #551](https://github.com/BrainJS/brain.js/issues/551)

## How Has This Been Tested?

I ran `npm run test` before and after my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Author's Checklist:

- [x] My code focuses on the main motivation and avoids scope creep.
- [x] My code passes current tests and adds new tests where possible.
- [x] My code is [SOLID](https://en.wikipedia.org/wiki/SOLID_(object-oriented_design)) and [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself).
- [x] I have updated the documentation as needed.

## Reviewer's Checklist:
- [ ] I kept my comments to the author positive, specific, and productive.
- [ ] I tested the code and didn't find any new problems.
- [ ] I think the motivation is good for the project.
- [ ] I think the code works to satisfies the motivation.

### CODE REVIEW

File: __tests__/istanbul-linker-util.ts [hunk 1]
```diff
@@ -1,18 +1,24 @@
...
 class Recurse {
-  constructor(settings) {
+  settings: TSettings;
+
+  constructor(settings: TSettings) {
     this.settings = settings;
   }
 
-  into(ast) {
+  into(ast: Node | any) {
```
Comment (mubaidr) [inline __tests__/istanbul-linker-util.ts]: I this `ast` should be optional here with type `Node`.

File: __tests__/istanbul-linker-util.ts [hunk 2]
```diff
@@ -136,14 +142,14 @@ class Recurse {
   }
 }
 
-module.exports = function (fn) {
+module.exports = function (fn: Object) {
```
Comment (mubaidr) [inline __tests__/istanbul-linker-util.ts]: `Object` is a generic, we may want a `function` definition here.

**Review by mubaidr — CHANGES_REQUESTED** [review]

File: __tests__/istanbul-linker-util.ts (hunk 1 above)
Comment (Nelias) [reply __tests__/istanbul-linker-util.ts]: There are no properties like 'type' existing on Node. In general istanbul is not even a typed library so this file should not be converted to TypeScript at all but if it is then `any` is what we have to use.

File: __tests__/istanbul-linker-util.ts (hunk 2 above)
Comment (Nelias) [reply __tests__/istanbul-linker-util.ts]: done

**Pull request merged**